prov/shm/src/smr.h

```diff
@@ -218,19 +218,21 @@ void smr_format_iov(struct smr_cmd *cmd, fi_addr_t peer_id,
 		void *context, struct smr_region *smr, struct smr_resp *resp,
 		struct smr_cmd *pend);
 
-int smr_tx_comp(struct smr_ep *ep, void *context, uint64_t flags, uint64_t err);
+int smr_tx_comp(struct smr_ep *ep, void *context, uint64_t flags,
+		uint16_t op_flags, uint64_t err);
 int smr_tx_comp_signal(struct smr_ep *ep, void *context, uint64_t flags,
-		       uint64_t err);
-int smr_rx_comp(struct smr_ep *ep, void *context, uint64_t flags, size_t len,
-		void *buf, void *addr, uint64_t tag, uint64_t data,
+		       uint16_t op_flags, uint64_t err);
+int smr_rx_comp(struct smr_ep *ep, void *context, uint64_t flags,
+		uint64_t msg_flags, size_t len, void *buf, void *addr,
+		uint64_t tag, uint64_t data,
 		uint64_t err);
 int smr_rx_src_comp(struct smr_ep *ep, void *context, uint64_t flags,
-		    size_t len, void *buf, void *addr, uint64_t tag,
-		    uint64_t data, uint64_t err);
-int smr_rx_comp_signal(struct smr_ep *ep, void *context, uint64_t flags,
+		    uint64_t msg_flags, size_t len, void *buf, void *addr,
+		    uint64_t tag, uint64_t data, uint64_t err);
+int smr_rx_comp_signal(struct smr_ep *ep, void *context, uint64_t flags, uint64_t msg_flags,
 		       size_t len, void *buf, void *addr, uint64_t tag,
 		       uint64_t data, uint64_t err);
-int smr_rx_src_comp_signal(struct smr_ep *ep, void *context, uint64_t flags,
+int smr_rx_src_comp_signal(struct smr_ep *ep, void *context, uint64_t flags, uint64_t msg_flags,
```

Comment:
same here for these calls.  Why double flags?

Reply:
Yeah, I thought I explained it enough in the commit message but I'll try to add something to clear it up or think of a way to consolidate them. They represent completion flags vs msg flags and the u64 vs u16 is because the tx flags come from the cmds which are trying to save space in the shared memory region so they get converted to an internal shm value which can be a u16) whereas the rx flags come from the rx_entry which is u64. I'll work on a solution to clear this up though.

Reply:
You mean I'm *supposed* to read the commit message?